smithy-aws-protocol-tests/model/restJson1/errors.smithy

```diff
@@ -31,13 +31,8 @@ apply GreetingWithErrors @httpResponseTests([
         documentation: "Ensures that operations with errors successfully know how to deserialize the successful response",
         protocol: restJson1,
         code: 200,
-        body: """
```

Comment:
TBH, I'm confused as to how any protocol test generators were able to pass this test case.

Reply:
I assume because none of the current protocol test implementors are _generating_ response bodies? So they probably just ignore the body since there are no fields in it for this shape?

Reply:
Looking at the v2 SDK's protocol test for this, it looks like it happily deserializes the response. I think I'm missing why generated tests should not be able to deserialize this response.

https://github.com/aws/aws-sdk-go-v2/blob/a55d57fffff519278eabd0e18ffc2bfeaac633c6/internal/protocoltest/awsrestjson/api_op_GreetingWithErrors_test.go#L30-L45

Reply:
Ah ! i see the greeting comes from a header, not the payload.

Reply:
Yeah, it's likely because deserialization code in clients just looks at the things it needs to look at for this test case (e.g., the header that greeting is bound to), and ignores the payload. I could see this kind of test being useful for pathological testing to ensure clients tolerate unexpected response content, but that should be in addition to expected responses, and they should always be marked with the `client-only` tag.